File: exercise/C#/day15/SantaChristmasList.Operations/Dependencies.cs
namespace SantaChristmasList.Operations;

public class Factory : Dictionary<Gift, ManufacturedGift>
{
    public ManufacturedGift FindManufacturedGift(Gift gift)
    {
        return ContainsKey(gift) ? this[gift] : null;
    }
}

public class Inventory : Dictionary<string, Gift>
{
    public Gift PickUpGift(string barCode)
    {
        return ContainsKey(barCode) ? this[barCode] : null;
    }
}

public class WishList : Dictionary<Child, Gift>
{
    public Gift IdentifyGift(Child child)
    {
        return ContainsKey(child) ? this[child] : null;
    }
}

File: exercise/ts/day15/src/dependencies.ts
import {Child, Gift, ManufacturedGift} from './models';

export class Factory extends Map<Gift, ManufacturedGift> {
    findManufacturedGift(gift: Gift): ManufacturedGift | undefined {
        return this.get(gift);
    }
}

export class Inventory extends Map<string, Gift> {
    pickUpGift(barCode: string): Gift | undefined {
        return this.get(barCode);
    }
}

export class WishList extends Map<Child, Gift> {
    identifyGift(child: Child): Gift | undefined {
        return this.get(child);
    }
}